tests/test_templatetags_provide.py

```diff
@@ -758,40 +894,125 @@ def get_template_data(self, args, kwargs, slots, context):
         template_str: types.django_html = """
             {% load component_tags %}
             {% provide "my_provide" key="hi" another=22 %}
-                {% component "injectee" %}
+                {% component "injectee22" %}
                 {% endcomponent %}
             {% endprovide %}
-            {% component "injectee" %}
+            {% component "injectee22" %}
             {% endcomponent %}
         """
         template = Template(template_str)
 
         with pytest.raises(KeyError):
             template.render(Context({}))
 
-        self._assert_clear_cache()
+        _assert_clear_cache()
+
+    # TODO - Enable once globals and finalizers are scoped to a single DJC instance")
+    #        See https://github.com/django-components/django-components/issues/1413
+    # @djc_test(parametrize=PARAMETRIZE_CONTEXT_BEHAVIOR)
+    @djc_test(
+        parametrize=(
+            ["components_settings"],
+            [
+                [{"context_behavior": "isolated"}],
+            ],
+            ["isolated"],
+        )
+    )
+    def test_inject_called_outside_rendering__persisted_ref(self, components_settings):
```

Comment:
And here are tests that verify that `Component.inject()` can be used even outside the render flow.